internal/resources/k8s_deployment_resource.go

```diff
@@ -603,3 +617,210 @@ func (r *KubernetesDeploymentResource) isProvisioning(tenantControlPlane *kamaji
+	}
+
+	return nil
+}
+
+func (r *KubernetesDeploymentResource) buildKonnectivityEgressSelectorFlag() string {
+	return fmt.Sprintf("--egress-selector-config-file=%s", konnectivityEgressSelectorConfigurationPath)
+}
+
+func (r *KubernetesDeploymentResource) addKonnectivityFlags(flags []string) {
+	flag := r.buildKonnectivityEgressSelectorFlag()
+	flags = append(flags, flag)
+}
+
+func (r *KubernetesDeploymentResource) addKonnectivityUDSVolume(configmapName string, v []corev1.Volume) {
+	volume := corev1.Volume{
+		Name: konnectivityUDSName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: configmapName,
+				},
+			},
+		},
+	}
+
+	v = append(v, volume)
+}
+
+func (r *KubernetesDeploymentResource) addKonnectivityUDSVolumeMount(v []corev1.VolumeMount) {
+	volume := corev1.VolumeMount{
+		Name:      konnectivityUDSName,
+		ReadOnly:  false,
+		MountPath: konnectivityServerPath,
+	}
+
+	v = append(v, volume)
+}
+
+func (r *KubernetesDeploymentResource) addKonnectivityServerContainer(containers []corev1.Container, tenantControlPlane kamajiv1alpha1.TenantControlPlane) {
+	container := r.getKonnectivityServerContainer(tenantControlPlane)
+	containers = append(containers, container)
+}
+
+func (r *KubernetesDeploymentResource) getKonnectivityServerContainer(tenantControlPlane kamajiv1alpha1.TenantControlPlane) corev1.Container {
+	// TODO: move to constants
+	return corev1.Container{
+		Name:    konnectivityServerName,
+		Image:   "us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-server:v0.0.16",
+		Command: []string{"/proxy-server"},
+		Args: []string{
+			"-v=8",
+			"--logtostderr=true",
+			fmt.Sprintf("--uds-name=%s/konnectivity-server.socket", konnectivityServerPath),
+			"--cluster-cert=/etc/kubernetes/pki/apiserver.crt",
+			"--cluster-key=/etc/kubernetes/pki/apiserver.key",
+			"--mode=grpc",
+			"--server-port=0",
+			"--agent-port=8132",
+			"--admin-port=8133",
+			"--health-port=8134",
+			"--agent-namespace=kube-system",
+			fmt.Sprintf("--agent-service-account=%s", tenantControlPlane.Status.Addons.Konnectivity.ServiceAccount.Name),
+			"--kubeconfig=/etc/kubernetes/konnectivity-server.conf",
+			fmt.Sprintf("--authentication-audience=%s", tenantControlPlane.Status.Addons.Konnectivity.ClusterRoleBinding.Name),
+		},
+		LivenessProbe: &corev1.Probe{
+			InitialDelaySeconds: 30,
+			TimeoutSeconds:      60,
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/healthz",
+					Port:   intstr.FromInt(8134),
+					Scheme: corev1.URISchemeHTTP,
+				},
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU: quantity.MustParse("100m"),
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "agentport",
+				ContainerPort: 8132,
```

Comment:
@mendrugory hard-coded port numbers for the proxy-server? The proxy-server port, default `8132` is expected to be exposed outside the cluster along with the api-server port, default to `6443`. The port should be reachable by the proxy-agent running on worker nodes in order to create the tunnel. It would be better to have the port configurable from the tcp manifest since users can have different requirements to expose a port.